### PR TITLE
Pubstwo 1601 advanced search bug

### PR DESCRIPTION
**Pull request by abriggs-usgs**
PUBSTWO-1602 Don't replicate series names

Added logic in utility module to filter out duplicates in lookupOptions array.

### CODE REVIEW

File: assets/scripts/pubswh/utils.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Takes an array of objects, checks for duplicate values of a given property, then
+ * creates a new array that excludes the objects containing the duplicated values
+ * @param {Array}  arrayWithDuplicates - an array of objects that have properties with duplicate values
+ * @param {String} propertyName - name of property to check for duplicate values
+ * @returns {Array} deduplicatedArray - an array of objects that have NO properties with duplicate values
+ */
+
+export function removeObjectsWithDuplicateValues(arrayWithDuplicates, propertyName) {
+    let deduplicatedArray = [];
+    for (let i = 0; i < arrayWithDuplicates.length; i++) {
```
Comment (mbucknell) [inline assets/scripts/pubswh/utils.js]: Consider using arrayWithDuplicates.forEach rather than a for loop

File: assets/scripts/pubswh/utils.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        // if the value for the current index is not in the new list that has no duplicates, function will return -1
+        let indexOfDuplicate = deduplicatedArray.findIndex(objectFromArray => objectFromArray.text === arrayWithDuplicates[i][propertyName]);
```
Comment (mbucknell) [inline assets/scripts/pubswh/utils.js]: I think objectFromArray.text should be objectFromArray[propertyName].

File: assets/scripts/pubswh/utils.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        // if the value is not in the new list that has no duplicates add the object containing that value to the list,
+        // otherwise do nothing
+            if (indexOfDuplicate === -1) {
```
Comment (mbucknell) [inline assets/scripts/pubswh/utils.js]: Not sure why this indented. Should be at the same indent level as line 13.

File: assets/tests/js/pubswh/utilsSpec.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import {removeObjectsWithDuplicateValues} from '../../../scripts/pubswh/utils';
+
+describe('Utils', function() {
+
+    var sampleArrayWithDuplicates = [{'id':1,'text':' duplicated listing','selected':false},{'id':2,'text':' duplicated listing','selected':false}];
+
+    it('expects if given an array with duplicates, it will return an array without duplicates', function () {
```
Comment (mbucknell) [inline assets/tests/js/pubswh/utilsSpec.js]: Consider adding a test for a list that contains no duplicates and adding a test for an empty array.

File: assets/tests/js/pubswh/utilsSpec.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+       expect(sampleArrayWithDuplicates.length).toEqual(2);
```
Comment (mbucknell) [inline assets/tests/js/pubswh/utilsSpec.js]: This is unnecessary. Please remove.

File: assets/scripts/pubswh/utils.js (hunk 1 above)
Comment (abriggs-usgs) [reply assets/scripts/pubswh/utils.js]: Good idea. Changed it.

File: assets/scripts/pubswh/utils.js (hunk 2 above)
Comment (abriggs-usgs) [reply assets/scripts/pubswh/utils.js]: Yup. You are correct.

File: assets/scripts/pubswh/utils.js (hunk 3 above)
Comment (abriggs-usgs) [reply assets/scripts/pubswh/utils.js]: Fixed.

File: assets/tests/js/pubswh/utilsSpec.js (hunk 4 above)
Comment (abriggs-usgs) [reply assets/tests/js/pubswh/utilsSpec.js]: More good advice. Done.

File: assets/tests/js/pubswh/utilsSpec.js (hunk 5 above)
Comment (abriggs-usgs) [reply assets/tests/js/pubswh/utilsSpec.js]: check:)

File: assets/tests/js/pubswh/utilsSpec.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    let sampleArrayWithDuplicates = [{'id':1,'text':' duplicated listing','selected':false},{'id':2,'text':' duplicated listing','selected':false}];
+    let sampleArrayWithoutDuplicates = [{'id':1,'text':' unique listing 1','selected':false},{'id':2,'text':' unique listing 2','selected':false}];
+    let sampleArrayEmpty = [];
+
+    it('expects if given an array with duplicates, it will return an array without duplicates', function() {
+       expect(removeObjectsWithDuplicateValues(sampleArrayWithDuplicates, 'text').length).toEqual(1);
+    });
+
+    it('expects if given an array with NO duplicates, it will return the same array', function() {
+        expect(removeObjectsWithDuplicateValues(sampleArrayWithoutDuplicates, 'text').length).toEqual(2);
```
Comment (mbucknell) [inline assets/tests/js/pubswh/utilsSpec.js]: Rather than call the function multiple times, you should assign the result to a variable.

Comment (abriggs-usgs) [reply assets/tests/js/pubswh/utilsSpec.js]: Oh. Yeah. Great Idea!
